### PR TITLE
Add ability to upload multiple data in one network request in NightscoutUploader

### PR DESCRIPTION
**Pull request by darinkrauss**
- Based off of latest `dev` branch
- Compatibility with latest LoopKit
- Additional functions for pull synchronization

### CODE REVIEW

File: NightscoutUploadKit/NightscoutUploader.swift [hunk 1]
```diff
@@ -396,6 +438,27 @@ public class NightscoutUploader {
...
+            completion(.success(false))
+            return
+        }
+
+        guard let url = url(for: .entries) else {
+            completion(.failure(UploadError.missingConfiguration))
+            return
+        }
+
+        postToNS(entries.map({$0.dictionaryRepresentation}), url: url) { result in
```
Comment (nhamming) [inline NightscoutUploadKit/NightscoutUploader.swift]: [nit] `.map({$0.dictionaryRepresentation})` vs `.map { $0.dictionaryRepresentation } as [Any]` above.

File: NightscoutUploadKit/NightscoutUploader.swift [hunk 2]
```diff
@@ -396,6 +438,27 @@ public class NightscoutUploader {
         }
     }
     
+    public func uploadEntries(_ entries: [NightscoutEntry], completion: @escaping (Result<Bool, Error>) -> Void) {
```
Comment (nhamming) [inline NightscoutUploadKit/NightscoutUploader.swift:428]: [nit] There seems to be redundant code in these new functions. Anyway to refactor?

**Review by nhamming — APPROVED** [review]
LGTM.

**Review by ps2 — APPROVED** [review]
LGTM!

File: NightscoutUploadKit/NightscoutUploader.swift (hunk 1 above)
Comment (darinkrauss) [reply NightscoutUploadKit/NightscoutUploader.swift]: @nhamming Would you please review the latest that includes changes for the above?

File: NightscoutUploadKit/NightscoutUploader.swift (hunk 2 above)
Comment (darinkrauss) [reply NightscoutUploadKit/NightscoutUploader.swift:428]: @nhamming Would you please review the latest that includes changes for the above?

**Comment by darinkrauss** [comment]
@ps2 @nhamming Please review latest updates responding to review comments.

**Review by nhamming — APPROVED** [review]
LGTM :shipit:

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**